aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1659,65 +1672,89 @@ AT_ERROR("cholesky: MAGMA library not found in "
     // which concisely is equal to batch_size % batch_limit
     if (batch_size % batch_limit != 0) {
       magmaCholeskyBatched<scalar_t>(
```

Comment:
nit, but consider bringing this to the upper loop, the only change needed is `mini_idx < batch_size` in the loop statement, and `nbatches = std::min(batch_limit, batch_size - mini_idx)` as argument to magmaCholeskyBatched.